app/controllers/stories/tagged_articles_controller.rb

```diff
@@ -30,45 +31,45 @@ def index
 
     private
 
-    def set_number_of_articles
-      @num_published_articles = if @tag_model.requires_approval?
-                                  @tag_model.articles.published.approved.count
+    def set_number_of_articles(tag:)
+      @num_published_articles = if tag.requires_approval?
+                                  tag.articles.published.approved.count
                                 elsif Settings::UserExperience.feed_strategy == "basic"
-                                  tagged_count
+                                  tagged_count(tag: tag)
                                 else
-                                  Rails.cache.fetch("article-cached-tagged-count-#{@tag}", expires_in: 2.hours) do
-                                    tagged_count
+                                  Rails.cache.fetch("article-cached-tagged-count-#{tag.name}", expires_in: 2.hours) do
+                                    tagged_count(tag: tag)
                                   end
                                 end
 
       @number_of_articles = user_signed_in? ? 5 : SIGNED_OUT_RECORD_COUNT
     end
 
-    def set_stories
-      @stories = Articles::Feeds::Tag.call(@tag, number_of_articles: @number_of_articles, page: @page)
+    def set_stories(number_of_articles:, page:, tag:)
+      stories = Articles::Feeds::Tag.call(tag.name, number_of_articles: number_of_articles, page: page)
 
-      @stories = @stories.approved if @tag_model&.requires_approval
+      stories = stories.approved if tag.requires_approval?
 
-      @stories = stories_by_timeframe
-      @stories = @stories.decorate
+      stories = stories_by_timeframe(stories: stories)
+      @stories = stories.decorate
     end
 
-    def tagged_count
-      @tag_model.articles.published.where("score >= ?", Settings::UserExperience.tag_feed_minimum_score).count
+    def tagged_count(tag:)
+      tag.articles.published.where("score >= ?", Settings::UserExperience.tag_feed_minimum_score).count
     end
 
-    def not_found_if_not_established
-      not_found if @stories.none? && !@tag_model.supported
+    def not_found_if_not_established(stories:, tag:)
+      not_found if stories.none? && !tag.supported?
     end
 
-    def stories_by_timeframe
+    def stories_by_timeframe(stories:)
       if %w[week month year infinity].include?(params[:timeframe])
-        @stories.where("published_at > ?", Timeframe.datetime(params[:timeframe]))
+        stories.where("published_at > ?", Timeframe.datetime(params[:timeframe]))
           .order(public_reactions_count: :desc)
       elsif params[:timeframe] == "latest"
-        @stories.where("score > ?", -20).order(published_at: :desc)
+        stories.where("score > ?", -20).order(published_at: :desc)
       else
-        @stories.order(hotness_score: :desc).where("score >= ?", Settings::UserExperience.home_feed_minimum_score)
+        stories.order(hotness_score: :desc).where("score >= ?", Settings::UserExperience.home_feed_minimum_score)
```

Comment:
We can use modern AR syntax here where an endless range is converted to a `>=` query.

```suggestion
        stories.order(hotness_score: :desc).where(score: Settings::UserExperience.home_feed_minimum_score..)
```